scarlet/lite/component.py

```diff
@@ -0,0 +1,621 @@
+    dSigmaX = ellipse.grad_minor(_grad, False)
+    dTheta = ellipse.grad_theta(_grad, False)
+    return np.array([dY0, dX0, dSigmaY, dSigmaX, dTheta, dN], dtype=params.dtype)
+
+
+class ParametricComponent:
+    """A parametric model
+    """
+
+    def __init__(self, sed, morph_params, morph_func, morph_grad, morph_prox,
+                 morph_step, model_frame, bbox, prox_sed=None, floor=1e-20):
+        """Initialize the component
+
+        Parameters
+        ----------
+        sed: `numpy.ndarray`
+            The SED of the component.
+        morph_params: `numpy.ndarray`
+            The parameters of the morphology.
+        morph_func: `Function`
+            The function to generate the morpholgy
+            based on `morphParams`.
+        morph_grad: `Function`
+            The function to calculate the gradient of the
+            likelihood wrt the morphological parameters.
+        morph_prox: `Function`
+            The proxmial operator for the morphology parameters.
+        bbox: `scarlet.bbox.Box`
+            The bounding box that holds the model.
+        frame: `CartesianGrid`
+            The coordinates of the model frame,
+            used to speed up the creation of the
+            polar grid for each source.
+        prox_sed: `Function`
+            Proximal operator for the SED.
+            If `prox_sed` is `None` then the default proximal
+            operator `self.prox_sed` is used.
+        floor: `float`
+            The minimum value of the SED, used to prevent
+            divergences in the gradients.
+        """
+        params = FixedParameter(morph_params)
+        sed = FixedParameter(sed)
+
+        self._params = params
+        self._func = morph_func
+        self._morph_grad = morph_grad
+        self._morph_prox = morph_prox
+        self._morph_step = morph_step
+        self._sed = sed
+        self._bbox = bbox
+        if prox_sed is None:
+            self._prox_sed = self.prox_sed
+        else:
+            self._prox_sed = prox_sed
+        self.slices = overlapped_slices(model_frame.bbox, bbox)
+        self.floor = floor
+
+    @property
+    def center(self):
+        """The center of the component"""
+        return self.y0, self.x0
+
+    @property
+    def y0(self):
+        """The y-center of the component"""
+        return self._params.x[0]
+
+    @property
+    def x0(self):
+        """The x-center of the component"""
+        return self._params.x[1]
+
+    @property
+    def sed(self):
+        """The SED of the component"""
+        return self._sed.x
+
+    @property
+    def bbox(self):
+        """The bounding box that contains the component"""
+        return self._bbox
+
+    @property
+    def frame(self):
+        """The coordinate system that contains the model"""
+        return CartesianFrame(self._bbox)
+
+    @property
+    def radial_params(self):
+        """The parameters used to model the radial function"""
+        return self._params.x
```

Comment:
Yes, mainly because in a circularly symmetric reference frame (which is the one in which the base `ParametricComponent` exists in), all of the parameters are radial and it was just easier to include the position in that. But you'll notice that in `EllipticalParametricComponent` they are not.